internal/service/s3/bucket_analytics_configuration.go

```diff
@@ -158,6 +158,10 @@ func resourceBucketAnalyticsConfigurationPut(ctx context.Context, d *schema.Reso
 		return conn.PutBucketAnalyticsConfiguration(ctx, input)
 	}, errCodeNoSuchBucket)
 
+	if tfawserr.ErrMessageContains(err, errCodeInvalidArgument, "AnalyticsConfiguration is not valid, expected CreateBucketConfiguration") {
```

Comment:
I've seen error messages change before. AWS says they shouldn't be verbatim relied on. Using the shortest possible matching text _might_ help avoid future problems.

```suggestion
	if tfawserr.ErrMessageContains(err, errCodeInvalidArgument, "Configuration is not valid") {
```